src/main/java/org/prebid/server/auction/ExchangeService.java

```diff
@@ -961,7 +965,7 @@ private static void tryAddWinningBidByBidder(Bid bid, String bidder,
      */
     private Future<Map<Bid, CacheIdInfo>> toBidsWithCacheIds(
```

Comment:
I would recommend changing return type of this method to something like `Future<CacheResult>`, where `CacheResult` will be a new POJO holding `Map<Bid, CacheIdInfo>`, cache service errors and response time. This will allow to move error collection and response time calculation to this method and keep this information in one place instead of scattering it here and there.